craft_application/errors.py

```diff
@@ -237,3 +237,92 @@ def __init__(  # (too many arguments)
             reportable=reportable,
             retcode=retcode,
         )
+
+
+class UbuntuProError(CraftError):
+    """Base Exception class for ProServices."""
+
+
+class UbuntuProApiError(UbuntuProError):
+    """Base class for exceptions raised during Ubuntu Pro Api calls."""
+
+
+class InvalidUbuntuProStateError(UbuntuProError):
+    """Base class for exceptions raised during Ubuntu Pro validation."""
+
+    # TODO: some of the resolution strings may not sense in a managed
+    # environment. What is the best way to get the is_managed method here?
+
+
+class UbuntuProClientNotFoundError(UbuntuProApiError):
+    """Raised when Ubuntu Pro client was not found on the system."""
+
+    def __init__(self, path: str) -> None:
+
+        message = f'The Ubuntu Pro client was not found on the system at "{path}"'
+
+        super().__init__(message=message)
```

Comment:
Please add a `resolution` and, if possible, a `docs_url` to this. The resolution can be as simple as "build without the --pro option."

Reply:
Closing this for now for the POC. We'll revisit before merging to main.